core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/TxnStatusFile.java

```diff
@@ -13,41 +13,70 @@
 
 import org.eclipse.rdf4j.common.io.NioFile;
 
+import static java.nio.charset.StandardCharsets.US_ASCII;
+
 /**
  * Writes transaction statuses to a file.
  */
 class TxnStatusFile {
+	public static final byte NONE_BYTE = (byte) 0b00000001;
+	public static final byte ACTIVE_BYTE = (byte) 0b00000010;
+	public static final byte COMMITTING_BYTE = (byte) 0b00000100;
+	public static final byte ROLLING_BACK_BYTE = (byte) 0b00001000;
+	public static final byte UNKNOWN_BYTE = (byte) 0b00010000;
 
 	public static enum TxnStatus {
 
 		/**
 		 * No active transaction. This occurs if no transaction has been started yet, or if all transactions have been
 		 * committed or rolled back.
 		 */
-		NONE,
+		NONE(NONE_BYTE),
 
 		/**
 		 * A transaction has been started, but was not yet committed or rolled back.
 		 */
-		ACTIVE,
+		ACTIVE(ACTIVE_BYTE),
 
 		/**
 		 * A transaction is being committed.
 		 */
-		COMMITTING,
+		COMMITTING(COMMITTING_BYTE),
 
 		/**
 		 * A transaction is being rolled back.
 		 */
-		ROLLING_BACK,
+		ROLLING_BACK(ROLLING_BACK_BYTE),
 
 		/**
 		 * The transaction status is unknown.
 		 */
-		UNKNOWN;
+		UNKNOWN(UNKNOWN_BYTE);
+
+		byte[] onDisk;
+
+		TxnStatus(byte onDisk) {
+			this.onDisk = new byte[1];
+			this.onDisk[0] = onDisk;
+		}
+
 	}
 
-	private static final Charset US_ASCII = Charset.forName("US-ASCII");
+	// THIS CODE BLOCK CAN BE REMOVED IN RDF4J release 4.0!!!
+	// check that none of the new TxnStatus conflict with the old ones, ie. if the first byte for the old values are the
+	// same as the first byte for the new values
+	{
+		boolean assertsEnabled = TxnStatus.class.desiredAssertionStatus();
+		if (assertsEnabled) {
+			for (TxnStatus value1 : TxnStatus.values()) {
+				for (TxnStatus value2 : TxnStatus.values()) {
+					byte firstByteInOldValue = value1.name().getBytes(US_ASCII)[0];
+					byte firstByteInNewValue = value2.onDisk[0];
+					assert firstByteInOldValue != firstByteInNewValue;
```

Comment:
I don't quite understand what you're checking here. It looks as if it's testing that the new byte representations have no overlaps with "legacy" txn statuses. Why do we need to run that test as part of this code? Surely you can make sure of that once (the status enums are constants, after all)?

Reply:
I moved it into a test instead. It checks that the first byte in the old statuses isn't the same as the first byte in the new ones.